src/com/cft/contactmerge/io/tests/SplitStringUtilitiesTest.java

```diff
@@ -3,16 +3,196 @@
 import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.*;
 import com.cft.contactmerge.io.*;
+import java.util.*;
 
 class SplitStringUtilitiesTest {
 
+    private String createVerificationMessage(String itemName)
+    {
+        return String.format("Verify %s", itemName);
+    }
+
+    /* ----------------------------------------------------------
+     * TSV Tests
+     * ---------------------------------------------------------- */
+    @Test
+    void splitTsvString_EmptyString() {
+        List<String> parts = SplitStringUtilities.splitTsvString("");
+
+        assertEquals(1, parts.size(), createVerificationMessage("result size"));
+        assertEquals( "", parts.iterator().next(), createVerificationMessage("first result"));
+    }
+
     @Test
-    void splitTsvString() {
-        // TODO: Add tests
+    void splitTsvString_SingleValue() {
+        List<String> parts = SplitStringUtilities.splitTsvString("Joe");
+
+        assertEquals(1, parts.size(), createVerificationMessage("result size"));
+        assertEquals( "Joe", parts.iterator().next(), "first result");
     }
 
     @Test
-    void splitCsvString() {
-        // TODO: Add tests
+    void splitTsvString_MultipleValues() {
+        List<String> expectedParts = Arrays.asList("Joe", "Smith", "123 Main St");
+        String testString = String.join("\t", expectedParts);
+
+        List<String> actualParts = SplitStringUtilities.splitTsvString(testString);
+
+        assertEquals(3, actualParts.size(), createVerificationMessage("result size"));
+
+        for(int i=0; i < actualParts.size(); i++) {
+            assertEquals(expectedParts.get(i), actualParts.get(i), String.format("result[%d]", i));
+        }
     }
+
+    @Test
+    void splitTsvString_MissingValues() {
+        List<String> expectedParts = Arrays.asList("", "Smith", "", "");
+        String testString = String.join("\t", expectedParts);
+
+        List<String> actualParts = SplitStringUtilities.splitTsvString(testString);
+
+        assertEquals(4, actualParts.size(), createVerificationMessage("result size"));
+
+        for(int i=0; i < actualParts.size(); i++) {
+            assertEquals(expectedParts.get(i), actualParts.get(i), String.format("result[%d]", i));
+        }
+    }
+
+    @Test
+    void splitTsvString_LeadingTrailingSpacesTrimmed() {
```

Comment:
Shouldn't these utility functions be "garbage in, garbage out", and trimming the the white space was going to be left to another function?

Reply:
Wow this is so cool

Reply:
I remember saying that the compare returns need to assume that the contact information was populated correctly. I don't think that has any relation to these methods though. These methods are the first ones to look at input data which can definitely be garbage. Their job is to recognize garbage and either fix it or reject it.